tests/e2e/lib/dpa_helpers.go

```diff
@@ -84,6 +98,73 @@ func (v *DpaCustomResource) Build(backupRestoreType BackupRestoreType) error {
 			},
 		},
 	}
+	switch v.Provider {
+	case "aws":
+		if v.OpenshiftCi {
+			if dpaInstance.Spec.BackupLocations[0].Velero.Config != nil {
+				dpaInstance.Spec.BackupLocations[0].Velero.Config["credentialsFile"] = "bsl-cloud-credentials-aws/cloud"
+			}
+		} else {
+			dpaInstance.Spec.BackupLocations[0].Velero.Credential = &corev1.SecretKeySelector{
+				LocalObjectReference: corev1.LocalObjectReference{
+					Name: v.CredSecretRef,
+				},
+				Key: "cloud",
+			}
+		}
+	case "gcp":
+		if v.OpenshiftCi {
+			if dpaInstance.Spec.BackupLocations[0].Velero.Config != nil {
+				dpaInstance.Spec.BackupLocations[0].Velero.Config["credentialsFile"] = "bsl-cloud-credentials-gcp/cloud"
+			}
+		} else {
+			dpaInstance.Spec.BackupLocations[0].Velero.Credential = &corev1.SecretKeySelector{
+				LocalObjectReference: corev1.LocalObjectReference{
+					Name: v.CredSecretRef,
+				},
+				Key: "cloud",
+			}
+		}
+		// dpaInstance.Spec.Configuration.Velero.DefaultPlugins = append(dpaInstance.Spec.Configuration.Velero.DefaultPlugins, oadpv1alpha1.DefaultPluginGCP)
+		// dpaInstance.Spec.SnapshotLocations = []oadpv1alpha1.SnapshotLocation{
+		// 	{
+		// 		Velero: &velero.VolumeSnapshotLocationSpec{
+		// 			Provider: v.Provider,
+		// 			Config: map[string]string{
+		// 				"snapshotLocation": v.gcpConfig.VslRegion,
+		// 			},
+		// 		},
+		// 	},
+		// }
+	case "azure":
+		dpaInstance.Spec.BackupLocations[0].Velero.Config = map[string]string{
+			"subscriptionId":          v.DpaAzureConfig.BslSubscriptionId,
+			"storageAccount":          v.DpaAzureConfig.BslStorageAccount,
+			"resourceGroup":           v.DpaAzureConfig.BslResourceGroup,
+			"storageAccountKeyEnvVar": v.DpaAzureConfig.BslStorageAccountKeyEnvVar,
+		}
+		if v.OpenshiftCi {
+			dpaInstance.Spec.BackupLocations[0].Velero.Config["credentialsFile"] = "bsl-cloud-credentials-azure/cloud" // <secret_name>/<key>
+		} else {
+			dpaInstance.Spec.BackupLocations[0].Velero.Credential = &corev1.SecretKeySelector{
+				LocalObjectReference: corev1.LocalObjectReference{
+					Name: v.CredSecretRef,
+				},
+				Key: "cloud",
+			}
+		}
+		dpaInstance.Spec.SnapshotLocations = []oadpv1alpha1.SnapshotLocation{
```

Comment:
is it possible to feed this info to the json instead? this way the code above already covers this initialization:
```
                        SnapshotLocations: v.CustomResource.Spec.SnapshotLocations, <<<<<<<<<<<<<<<<<<<<<
			BackupLocations: []oadpv1alpha1.BackupLocation{
				{
					Velero: &velero.BackupStorageLocationSpec{
						Provider:   v.CustomResource.Spec.BackupLocations[0].Velero.Provider,
						Default:    true,
						Config:     v.CustomResource.Spec.BackupLocations[0].Velero.Config,
						Credential: v.CustomResource.Spec.BackupLocations[0].Velero.Credential,
```